### PR TITLE
Introduce type-only imports

### PR DESCRIPTION
**Pull request by smashercosmo**
**Summary**
Introduce type-only imports and enable @typescript-eslint/consistent-type-imports eslint rule (part of https://github.com/netlify/netlify-cms/pull/5374)

**Test plan**
Current tests

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -78,11 +78,13 @@ module.exports = {
         },
       },
       rules: {
-        'require-atomic-updates': [0],
-        'import/no-named-as-default': 0,
+        'no-duplicate-imports': [0], // handled by @typescript-eslint
+        '@typescript-eslint/ban-types': [0], // TODO enable in future
```
Comment (smashercosmo) [inline .eslintrc.js:82]: This rule forbids using empty object as type. Probably good idea to enable it in future.

File: packages/netlify-cms-backend-bitbucket/src/API.ts [hunk 2]
```diff
@@ -653,8 +653,8 @@ export default class API {
       params: {
         pagelen: 50,
         q: oneLine`
-        source.repository.full_name = "${this.repo}" 
-        AND state = "${BitBucketPullRequestState.OPEN}" 
+        source.repository.full_name = "${this.repo}"
+        AND state = "${BitBucketPullRequestState.OPEN}"
```
Comment (smashercosmo) [inline packages/netlify-cms-backend-bitbucket/src/API.ts:657]: Hm... I guess this will break. We need to enforce spaces here somehow.

Comment (erezrokah) [reply packages/netlify-cms-backend-bitbucket/src/API.ts:657]: Let me look into that. I'm mostly concerned that this doesn't happen when I run `yarn format`

Comment (smashercosmo) [reply packages/netlify-cms-backend-bitbucket/src/API.ts:657]: Hm... I does happen when I run ```yarn format```

Comment (smashercosmo) [reply packages/netlify-cms-backend-bitbucket/src/API.ts:657]: Oh, no, I was wrong. This is something my IDE is doing on save. Weird.

Comment (erezrokah) [reply packages/netlify-cms-backend-bitbucket/src/API.ts:657]: That's very odd as we also run prettier in CI on Windows, MacOS and Ubuntu and it passes.

Maybe something specific to your machine?

Regardless, they are redundant so I'll delete them in another PR and fix the recorded data [here](https://github.com/netlify/netlify-cms/tree/e4a29d5991f648b98b7df9f1a9a8c6448560851b/cypress/fixtures)

Comment (erezrokah) [reply packages/netlify-cms-backend-bitbucket/src/API.ts:657]: Well actually looks like tests are passing so it seems `oneLine` trims those so we don't need to update any recorded data.

Comment (erezrokah) [reply packages/netlify-cms-backend-bitbucket/src/API.ts:657]: > Oh, no, I was wrong. This is something my IDE (WebStorm) is doing on save. Weird.

Just noticing this, I still think we can remove those

**Review by erezrokah — APPROVED** [review]

**Pull request merged**